### PR TITLE
adding export button

### PR DESCRIPTION
**Pull request by parteekcoder**
issue #366 

hey @llaske I am making functionality of exporting assignment , yet I am making this  when it is completed I will inform you

### CODE REVIEW

**Comment by parteekcoder** [comment]
@llaske please review this PR , I know there are some changes required to include necessary fields in the CSV file

I am open to new suggestions and Ideas , your suggestions will be admired 

Please tell me if there is anything you would like to update , I have tested this on my system

File: dashboard/controller/assignments/exportCSV.js [hunk 1]
```diff
@@ -0,0 +1,92 @@
+// include libraries
+var superagent = require('superagent'),
+	async = require('async'),
+	common = require('../../helper/common');
+
+module.exports =  function exportCSV(req, res) {
+	var assignment = [];
+
+	const validateAssignment=(assignment)=>{
+		const assign={
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportCSV.js:10]: Sugarizer Server code follows ES5 syntax, so it better to avoid ES6 syntax.

File: dashboard/controller/assignments/exportCSV.js [hunk 2]
```diff
@@ -0,0 +1,92 @@
+// include libraries
+var superagent = require('superagent'),
+	async = require('async'),
+	common = require('../../helper/common');
+
+module.exports =  function exportCSV(req, res) {
+	var assignment = [];
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportCSV.js:7]: The variable name conflicts with the other variables in the same function, for example `assignment` on line 9.

File: dashboard/controller/assignments/exportCSV.js [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+
+	const validateAssignment=(assignment)=>{
+		const assign={
+			_id:"",
+			name:"",
+			title:"",
+			activity:"",
+			creation_time:"",
+			stroke:"",
+			fill:"",
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportCSV.js:17]: The fill and stroke information is not relevant to delivery.

File: dashboard/controller/assignments/exportCSV.js [hunk 4]
```diff
@@ -0,0 +1,92 @@
...
+		return assign;
+
+	}
+	async.series([
+		function(callback) {
+			superagent
+				.get(common.getAPIUrl(req) + 'api/v1/assignments')
+				.set(common.getHeaders(req))
+				.end(function (error, response) {
+					// console.log(response.body);
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportCSV.js:69]: Remove unnecessary comments

File: dashboard/controller/assignments/exportCSV.js [hunk 5]
```diff
@@ -0,0 +1,92 @@
...
+		if(assignment.isAssigned){
+			assign.isAssigned=assignment.isAssigned;
+		}
+		return assign;
+
+	}
+	async.series([
+		function(callback) {
+			superagent
+				.get(common.getAPIUrl(req) + 'api/v1/assignments')
```
Comment (NikhilM98) [inline dashboard/controller/assignments/exportCSV.js:66]: I don't think the issue is about exporting Assignments. The requirement is for exporting the Deliveries.

File: dashboard/views/assignments.ejs [hunk 6]
```diff
@@ -127,7 +179,7 @@
                                                             <a data-l10n-id="editAssignment" style=" cursor: not-allowed; pointer-events: none;" title="Edit Assignment" href="/dashboard/assignments/edit/<%= data.assignments[i]._id %>"><i style="color:#ddd" id="assignment-edit-class" class="material-icons text-muted">edit</i></a> 
                                                             <a data-l10n-id="deleteAssignment" style=" cursor: not-allowed; pointer-events: none;" title="Delete Assignment" href="/dashboard/assignments/delete/<%= data.assignments[i]._id %>?&name=<%= data.assignments[i].name %>" onclick="return confirm(document.webL10n.get('DoYouWantDeleteAssignment', {assignment:'<%= data.assignments[i].name %>'}))"><i style="color:#ddd" id="assignments-delete-class" class="material-icons text-muted">delete_forever</i></a>
                                                         <% } %> 
-                                                    <% } %>
+                                                    <% } %>
```
Comment (NikhilM98) [inline dashboard/views/assignments.ejs:182]: Remove unnecessary whitespace.

File: dashboard/views/assignments.ejs [hunk 7]
```diff
@@ -148,4 +200,7 @@
     handleSort();
     var currTour = sugarizerTour("assignment", <% if (account.user && account.user.role=="admin") {"admin"} else {"teacher"} %>);
     if (!currTour.isFinished()) currTour.start();
+
+    //function for exporting assigment
+
```
Comment (NikhilM98) [inline dashboard/views/assignments.ejs:205]: Remove unnecessary comments.

**Review by NikhilM98 — CHANGES_REQUESTED** [review]
I believe you have misunderstood the requirements of the issue. The requirement is to export the deliveries, not the list of assignments.

File: dashboard/controller/assignments/exportCSV.js (hunk 2 above)
Comment (parteekcoder) [reply dashboard/controller/assignments/exportCSV.js:7]: no that will work because of scope in javascript

File: dashboard/controller/assignments/exportCSV.js (hunk 3 above)
Comment (parteekcoder) [reply dashboard/controller/assignments/exportCSV.js:17]: sure

File: dashboard/controller/assignments/exportCSV.js (hunk 4 above)
Comment (parteekcoder) [reply dashboard/controller/assignments/exportCSV.js:69]: yeah I will create a request to assugnment delivers as you describe me

File: dashboard/controller/assignments/exportCSV.js (hunk 2 above)
Comment (NikhilM98) [reply dashboard/controller/assignments/exportCSV.js:7]: Yeah, it will work, but it's better to have a different name for clarity, for example `assignments` in this case.

**Comment by parteekcoder** [comment]
thanks @NikhilM98 for resolving my queries

**Comment by parteekcoder** [comment]
@NikhilM98 I am creating an assignment with a teacher account 

image
but I am not getting any activity option

**Comment by NikhilM98** [comment]
You can log into Sugarizer app with a teacher account and create a journal entry to make it visible here.

**Comment by parteekcoder** [comment]
> You can log into Sugarizer app with a teacher account and create a journal entry to make it visible here.

but that page is not opeining because the students don't have any journals, how a student can opt a journal

**Comment by NikhilM98** [comment]
Open localhost:8080 to access the Sugarizer Client.

**Comment by parteekcoder** [comment]
image

but here a student will opt an journal

**Comment by NikhilM98** [comment]
Create a teacher account from Sugarizer Server dashboard, and log into the Sugarizer Client with the teacher account.

**Comment by parteekcoder** [comment]
yeah done , but I am not able to give journal to student or creating them, So that they will show me while adding assignemnt

**Comment by parteekcoder** [comment]
image

**Comment by NikhilM98** [comment]
Looks like your Teacher's journal is empty. Try using Sugarizer Activities, it will create Journal entries. You can assign those activities as an Assignment.

**Comment by parteekcoder** [comment]
how , can you please give some steps to do that

**Comment by parteekcoder** [comment]
As there is no Docs so it is difficult to do

**Comment by parteekcoder** [comment]
journals are showing
image

**Comment by parteekcoder** [comment]
but still while creating asssignment that activity option is still not showing

**Comment by parteekcoder** [comment]
I want to test some things so I need to create some assignments @NikhilM98 please help

**Comment by NikhilM98** [comment]
Now try to log into Sugarizer Server dashboard with the teacher's account and try to create an assignment.

**Comment by parteekcoder** [comment]
Yeah doing that but still activities are not showing while creating assignment

**Comment by NikhilM98** [comment]
I just tested it with a fresh Sugarizer Server instance. It's working: 
image

You must be doing something wrong.

**Comment by NikhilM98** [comment]
> journals are showing image

Btw these journal entries do not belong to the user you have logged in with.

**Comment by NikhilM98** [comment]
> As there is no Docs so it is difficult to do

We expect potential contributors to learn to use the application by experimenting with the application. While we understand that this approach may be challenging, it is also the best way to gain a thorough understanding of the application's functionality and how it is used in practice.

We do not want to provide a step-by-step documentation for the developers as we believe that the most valuable contributions come from those who have taken the time to really explore and understand the application on their own. We encourage you to spend some time using the application, exploring its various features, and familiarizing yourself with its functionality. This will not only help you gain a better understanding of the application, but it will also help you identify areas where you can make valuable contributions.

**Comment by parteekcoder** [comment]
> > As there is no Docs so it is difficult to do
> 
> We expect potential contributors to learn to use the application by experimenting with the application. While we understand that this approach may be challenging, it is also the best way to gain a thorough understanding of the application's functionality and how it is used in practice.
> 
> We do not want to provide a step-by-step documentation for the developers as we believe that the most valuable contributions come from those who have taken the time to really explore and understand the application on their own. We encourage you to spend some time using the application, exploring its various features, and familiarizing yourself with its functionality. This will not only help you gain a better understanding of the application, but it will also help you identify areas where you can make valuable contributions.

Right @NikhilM98 I agree with your statement that `the best contributor is the one who explore the things himself/herself`, But the thing is docs help the contributors to initially setup and run all the things and the rest of the things they can explore

**Comment by NikhilM98** [comment]
Docs for the initial deployment of Sugarizer Server are available [here](https://github.com/llaske/sugarizer-server/tree/master/docs). PRs are welcome if you believe that the docs can be improved or the scope of the docs can be extend.

**Comment by parteekcoder** [comment]
@NikhilM98 I created my first assignment
image

**Comment by NikhilM98** [comment]
@parteekcoder can this PR be closed?

**Comment by parteekcoder** [comment]
yes , continued at #385
